### PR TITLE
Migrate to pyproject.toml

### PR DESCRIPTION
**Pull request by mshriver**
Remove entry-points, setup.cfg, setup.py, requirements.txt, test-requirements, tox.ini, ruff.toml files

Replace with pyproject.toml specifications

Move to hatchling for build backend

### CODE REVIEW

File: backend/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,106 @@
+[project]
+dynamic = ["version"]
+name = "Ibutsu Server"
+description = "A system to store and query test results and artifacts"
+authors = [
+    {name = "Raoul Snyman"},
```
Comment (mshriver) [inline backend/pyproject.toml:6]: @rsnyman hey! Is there an email that you would like me to add here? I'd like to keep you as the original author here.

File: backend/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,103 @@
+[project]
+version = "2.5.8"
+name = "ibutsu_server"
+description = "A system to store and query test results and artifacts"
+authors = [
+    {name = "Raoul Snyman"},
+    {name = "John Dupuy"},
+]
+maintainers = [
+    {name = "Mike Shriver", email = "mshriver@redhat.com"},
```
Comment (mshriver) [inline backend/pyproject.toml:10]: @RonnyPfannschmidt @jjaquish @LightOfHeaven1994  (and anyone else that sees this)

Would you like to be listed here? Red Hat emails?

Comment (RonnyPfannschmidt) [reply backend/pyproject.toml:10]: -1 for me

File: backend/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,106 @@
+[project]
+dynamic = ["version"]
+name = "Ibutsu Server"
+description = "A system to store and query test results and artifacts"
+authors = [
+    {name = "Raoul Snyman"},
+    {name = "John Dupuy"},
```
[the remark this answers is not in the document]
Comment (john-dupuy) [reply backend/pyproject.toml:7]: Howdy @mshriver - hope you've been well! Sure, you can add `jdupuy26@gmail.com`

File: backend/pyproject.toml (hunk 2 above)
Comment (LightOfHeaven1994) [reply backend/pyproject.toml:10]: @mshriver yep, add me please :pray:

File: scripts/ibutsu-pod.sh [hunk 4]
```diff
@@ -180,24 +180,24 @@ podman run -d \
        $PYTHON_IMAGE \
        /bin/bash -c 'pip3 install -U pip setuptools wheel &&
                      pip3 install -r requirements.txt &&
```
Comment (LightOfHeaven1994) [inline scripts/ibutsu-pod.sh:182]: no more setuptools and requirements

**Review by LightOfHeaven1994 — CHANGES_REQUESTED** [review]
Most of the things look good.  But I found some items that have to be changed. 

Also check `getting-started.rst` as it should be updated (remove requirements)

**Pull request merged**